src/main/java/de/srendi/advancedperipherals/client/screens/KeyboardScreen.java

```diff
@@ -0,0 +1,239 @@
+package de.srendi.advancedperipherals.client.screens;
+
+import com.mojang.blaze3d.platform.InputConstants;
+import com.mojang.blaze3d.platform.Window;
+import com.mojang.blaze3d.vertex.PoseStack;
+import dan200.computercraft.client.gui.ClientInputHandler;
+import dan200.computercraft.client.gui.widgets.WidgetTerminal;
+import dan200.computercraft.core.terminal.Terminal;
+import dan200.computercraft.shared.computer.core.InputHandler;
+import de.srendi.advancedperipherals.client.ClientWorker;
+import de.srendi.advancedperipherals.common.container.KeyboardContainer;
+import de.srendi.advancedperipherals.common.network.APNetworking;
+import de.srendi.advancedperipherals.common.network.toserver.KeyboardMouseClickPacket;
+import de.srendi.advancedperipherals.common.network.toserver.KeyboardMouseMovePacket;
+import de.srendi.advancedperipherals.common.network.toserver.KeyboardMouseScrollPacket;
+import net.minecraft.client.KeyMapping;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.MenuAccess;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Inventory;
+import org.jetbrains.annotations.NotNull;
+import org.lwjgl.glfw.GLFW;
+
+/**
+ * A simple screen but without any rendering calls. Used to unlock the mouse so we can freely write stuff
+ * <p>
+ * We just create a terminal which is used to forward all the key presses and mouse clicks but we don't render it.
+ */
+public class KeyboardScreen extends Screen implements MenuAccess<KeyboardContainer> {
+
+    protected final KeyboardContainer keyboardContainer;
+    protected final InputHandler input;
+    private final Terminal terminalData;
+    private WidgetTerminal terminal;
+    private MouseState mouseState = MouseState.RELEASED;
+    private boolean captureMouse;
+    private boolean regrabingMouse;
+    private byte[] lastPosLock = new byte[0];
```

Comment:
```suggestion
    private final byte[] lastPosLock = new byte[0];
```
I forgot make this final so it confused me a bit